src/pages/settings/tax-settings/components/calculate-taxes/components/EditSubRegionModal.tsx

```diff
@@ -0,0 +1,82 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { Button, InputField } from '$app/components/forms';
+import { Modal } from '$app/components/Modal';
+import { ChangeEvent, Dispatch, SetStateAction } from 'react';
+import { useTranslation } from 'react-i18next';
+import { TaxSetting } from '$app/common/interfaces/company.interface';
+import { useHandleCurrentCompanyChangeProperty } from '$app/pages/settings/common/hooks/useHandleCurrentCompanyChange';
+
+interface Props {
+    region: string;
+    subregion: string;
+    visible: boolean;
+    setVisible: Dispatch<SetStateAction<boolean>>;
+    taxSetting: TaxSetting;
+}
+
+export function EditSubRegionModal(props: Props) {
+    const [t] = useTranslation();
+    
+    const {
+        visible,
+        setVisible,
+        region,
+        subregion,
+        taxSetting,
+    } = props;
```

Comment:
If you're fan of this, I'd suggest destructuring on the fn signature.

```tsx
export function EditSubRegionModal({visible, setVisible, etc}: Props)
```

Reply:
need assistanace on this one.